File: solutions/r/perfect-numbers/1/perfect-numbers.R
factoring_sum = function(n) {
  vetor = c();
  for (i in seq(1,n-1)){
    if (n %% i == 0){
      vetor = c(vetor, i);
    }
  }
  return (sum(vetor))
}

number_type = function(n) {
  if (n < 1){
    stop();
  } else if(n==1){
    return ("deficient")
  } else {
  soma = factoring_sum(n);
  if(n == soma){
    return ("perfect")
  }else if(soma > n){
    return ("abundant")
  } else {
    return ("deficient")
  }
}
}

File: solutions/python/perfect-numbers/1/perfect_numbers.py
def factoring_sum(number):
    vector = [];
    for i in range(1,number):
        if(number % i == 0):
            vector.append(i)
    return sum(vector)
def classify(number):
    if (number < 1):
        raise ValueError("Number must be positive")
    elif number == 1:
        return "deficient"
    else:
        facSum = factoring_sum(number);
        if number == facSum:
            return "perfect"
        elif facSum > number:
            return "abundant"
        else:
            return "deficient"